CMakeLists.txt

```
@@ -116,6 +116,10 @@ configure_file(${CMAKE_SOURCE_DIR}/tutorials.md.in ${CMAKE_BINARY_DIR}/tutorials
 ign_create_docs(
   API_MAINPAGE_MD "${CMAKE_BINARY_DIR}/api.md"
   TUTORIALS_MAINPAGE_MD "${CMAKE_BINARY_DIR}/tutorials.md"
+  AUTOGENERATED_DOC "${CMAKE_BINARY_DIR}/include/ignition/msgs/"
   TAGFILES
     "${IGNITION-MATH_DOXYGEN_TAGFILE} = ${IGNITION-MATH_API_URL}"
 )
+
+# wait to build the doc to the autogenerated code
+add_dependencies(doc ignition-msgs6)
```

```suggestion
add_dependencies(doc ignition-msgs${PROJECT_VERSION_MAJOR})
```

Let's avoid hardcoding the version because we always have a hard time updating these details when we bump the number.

---

This is also causing the compilation to fail in case doxygen is not present, see CI:

```
   -- Could NOT find Doxygen (missing: DOXYGEN_EXECUTABLE) 
    Cannot add target-level dependencies to non-existent target "doc".
```

Maybe this `add_dependencies` line needs to be added inside `ign-cmake`?

---

I added a condition to check if the target exists, this will avoid adding a another argument to `ign_create_docs`

https://github.com/ignitionrobotics/ign-msgs/blob/d592a22ca300e58b4b7091dd012be227641259fa/CMakeLists.txt#L124-L128